network/p2p/tracer/internal/rpc_sent_tracker_test.go

```diff
@@ -0,0 +1,71 @@
+package internal
+
+import (
+	"testing"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	pb "github.com/libp2p/go-libp2p-pubsub/pb"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/network/channels"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// TestNewRPCSentTracker ensures *RPCSenTracker is created as expected.
+func TestNewRPCSentTracker(t *testing.T) {
+	tracker := mockTracker(t)
+	require.NotNil(t, tracker)
+}
+
+// TestRPCSentTracker_IHave ensures *RPCSentTracker tracks sent iHave control messages as expected.
+func TestRPCSentTracker_IHave(t *testing.T) {
+	tracker := mockTracker(t)
+	require.NotNil(t, tracker)
+
+	t.Run("WasIHaveRPCSent should return false for iHave message Id that has not been tracked", func(t *testing.T) {
```

Comment:
can you also add a test for a message with the same topicID but different message id (possibly different last byte)?

Reply:
https://github.com/onflow/flow-go/pull/4543/commits/c7119f9d482369d0bd39517ad775e247294ff9c3